core-bundle/src/Resources/contao/models/PageModel.php

```diff
@@ -1166,7 +1166,7 @@ public function getPreviewUrl($strParams=null)
 		$baseUrl = $context->getBaseUrl();
 
 		// Add the preview script
-		$context->setBaseUrl($previewScript);
+		$context->setBaseUrl(\dirname($context->getBaseUrl()) . $previewScript);
```

Comment:
What issue does this fix?

Reply:
If you are logged into the back end via `https://example.com/foobar/public/` the preview links in the back end will be wrong otherwise.

Reply:
How do I reproduce the issue?

Reply:
I cannot reproduce this issue.

1. I open `https://localhost/managed-edition/public/contao`.
2. The page preview link is `https://localhost/managed-edition/public/contao/preview?page=51`.
3. If I click the link, the `BackendPreviewController` redirects me to `https://localhost/managed-edition/public/preview.php/contao/preview?page=52`.
4. On this page in the `PageModel::getPreviewUrl()` method, `$context->getBaseUrl()` is the same as `\dirname($context->getBaseUrl()) . $previewScript`, so the whole `$context->setBaseUrl()` call is redundant.

How exactly do I reproduce your case?

Reply:
In line 1156 we set `$previewScript`, which will always be `/preview.php` and in line 1169 we override the base URL with `$previewScript`. So at that point `$context->getBaseUrl()` will be wrong as it will be missing the original base. I am not sure why you are not able to reproduce the issue, this issue is definitely present in `5.x` and I can still reproduce it as described.

Reply:
I was also able to reproduce this, and this change fixed it.